src/App.tsx

```diff
@@ -32,24 +37,35 @@ const StyledRoutesContainer = styled(AppGrid)({
 });
 
 const App = () => {
-  const { restoreSession } = useWalletActions();
+  const { restoreSession, restoreAdapter } = useWalletActions();
+  const { adapterId } = useWalletStore();
+  const wallets = useSelector((state: RootState) => state.wallet.allWallets)
+  const dispatch = useDispatch<any>()
+  const { txPending } = useTokenOperationsStore();
+  const adapter = useSelectedAdapter()
+  const {t} = useTranslation()
   useWebAppResize();
 
-  useEffectOnce(() => {
-    restoreSession();
-  });
-
   useEffect(() => {
     (async () => {
       const _client = new TonClient({
         endpoint: "https://mainnet.tonhubapi.com/jsonRPC",
       });
       setClienT(_client)
     })();
+    dispatch(fetchTonConnectWallets())
+    restoreSession();
   }, [])
 
+  useEffect(() => {
```

Comment:
usually its better practice to do:
const walletsLength = wallets.length

useEffect(() => {
do something when length changes
}, [walletsLength])

Reply:
fixed